pkg/bloomcompactor/compress_utils.go

```diff
@@ -0,0 +1,79 @@
+package bloomcompactor
+
+import (
+	"fmt"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	v1 "github.com/grafana/loki/pkg/storage/bloom/v1"
+	"github.com/grafana/loki/pkg/storage/stores/shipper/bloomshipper"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func compressBloomBlock(storedBlock bloomshipper.Block, archivePath, localDst string, logger log.Logger) (bloomshipper.Block, error) {
```

Comment:
I think we should only pass the block's `BlockRef`, since `Data` is not used.

```suggestion
func compressBloomBlock(ref bloomshipper.BlockRef, archivePath, localDst string, logger log.Logger) (bloomshipper.Block, error) {
```